bdk_core/src/tx_graph.rs

```diff
@@ -127,6 +132,21 @@ impl TxGraph {
         })
     }
 
+    /// Iterate over all full transactions in the graph
+    pub fn iter_full_transactions(&self) -> impl Iterator<Item = &Transaction> {
+        self.txs.iter().filter_map(|(_, tx)| match tx {
+            TxNode::Whole(tx) => Some(tx),
+            TxNode::Partial(_) => None,
+        })
+    }
+
+    pub fn iter_partial_transactions(&self) -> impl Iterator<Item = (Txid, &BTreeMap<u32, TxOut>)> {
+        self.txs.iter().filter_map(|(txid, tx)| match tx {
+            TxNode::Whole(_) => None,
+            TxNode::Partial(partial) => Some((*txid, partial)),
+        })
+    }
```

Comment:
Note I don't think I actually need these right now but added them since it feels like that's a basic thing you might want to do.